src/parse.c

```diff
@@ -572,6 +572,29 @@ get_default_backend_for_type(NetplanDefType type)
     return NETPLAN_BACKEND_NETWORKD;
 }
 
+static gboolean
+handle_access_point_bool(yaml_document_t* doc, yaml_node_t* node, const void* data, GError** error)
+{
+    guint offset = GPOINTER_TO_UINT(data);
+    gboolean v;
+
+    if (g_ascii_strcasecmp(scalar(node), "true") == 0 ||
```

Comment:
I know that refactoring should not really be part of new functionality PRs, but here I feel like we might want to get rid of the redundancy. This bool checking is happening too frequently and has just copy-pasted code all the time. I'd look into splitting this out instead.

Reply:
Yep, I agree as mentioned in the main PR description. I'm happy to do the refactor either way - in this PR or in a separate one, just wasn't sure what the preference would be.

Reply:
I refactored this in 3aacdfc